src/foundry/common/abstract/type-data.d.mts

```diff
@@ -16,10 +19,9 @@ declare const _InternalTypeDataModelConst: _InternalTypeDataModelInterface;
 declare class _InternalTypeDataModel<
   Schema extends DataSchema,
   Parent extends Document<DataSchema, any, any>,
-  BaseData extends Record<string, unknown> = Record<string, never>,
   DerivedData extends Record<string, unknown> = Record<string, never>,
   // This does not work if inlined. It's weird to put it here but it works.
-  _ComputedInstance extends object = Merge<BaseData, DerivedData>,
+  _ComputedInstance extends object = DerivedData,
```

Comment:
This would be where the removal of the index signature would be correct; `_ComputedInstance = RemoveIndexSignature<...>`. I already pushed this to v10 a while back.

This would be `RemoveIndexSignature<Merge<BaseData, DerivedData>>` using my other comments, because again, the base data phase and the derived data phase are separate. This distinction _can_ be important.